tools/elf2uf2/main.cpp

```diff
@@ -226,7 +238,7 @@ int elf2uf2(FILE *in, FILE *out) {
     bool ram_style = false;
     address_ranges valid_ranges = {};
     if (!rc) {
-        ram_style = 0x2 == eh.entry >> 28u;
+        ram_style = is_address_initialized(rp2040_address_ranges_ram, eh.entry);
```

Comment:
@kilograham I guess this relies on there being no overlapping `address_range::type::CONTENTS` for any address covered by the ranges in `rp2040_address_ranges_flash` and `rp2040_address_ranges_ram`. Is it worth adding an assertion for that in `main()`, or would that just be overkill?

Reply:
yeah i think that is overkill